mlir/include/mlir/Dialect/Func/IR/FuncOps.td

```diff
@@ -157,6 +157,11 @@ def CallIndirectOp : Func_Op<"call_indirect", [
 
     /// Return the callee of this operation.
     CallInterfaceCallable getCallableForCallee() { return getCallee(); }
+
+    /// Set the callee for this operation.
+    void setCalleeFromCallable(CallInterfaceCallable callee) {
+      setOperand(0, dyn_cast<Value>(callee));
+    }
```

Comment:
```suggestion
    /// Set the callee for this operation.
    void setCalleeFromCallable(CallInterfaceCallable callee) {
      setOperand(0, dyn_cast<Value>(callee));
    }
```

I think this is a bit limiting, as `func.constant` does not implement `CallInterfaceCallable`. Maybe we can have two flavors: `setCallee` (receiving a `SymbolRefAttr`) and `setIndirectCallee` (receiving a `Value`). WDYT?

This way, e.g., for `func.call`, only the flavour receiving a `SymbolRefAttr` would be valid (the other will raise an error) and for `func.indirect_call`, it will be the other way around.

Reply:
I don't quite understand why `func.constant` does not implement `CallInterfaceCallable` causes this to be limiting. 
Shouldn't `setCalleeFromCallable` allow receiving both `SymbolRefAttr` and `Value`? And depends on the implementation for the operation, it could be either or both?

The reason we assume `callee` is a `Value` here, is because 
```
CallInterfaceCallable getCallableForCallee() { return getCallee(); }
::mlir::TypedValue<::mlir::FunctionType> CallIndirectOp::getCallee() {
  return ::llvm::cast<::mlir::TypedValue<::mlir::FunctionType>>(*getODSOperands(0).begin());
}
```